flanker/mime/message/headers/parametrized.py

```diff
@@ -107,8 +107,11 @@ def concatenate(parts):
     if is_old_style(part):
         # old-style parameters do not support any continuations
         return encodedword.mime_to_unicode(get_value(part))
-
-    return ''.join(decode_new_style(p) for p in partition(parts))
+    elif is_single_part(part):
```

Comment:
I had to noodle through all of this for a bit to understand what it was now doing. We should probably enhance the docstring for this function that includes these two new formats it now supports.